### PR TITLE
[Rust] Fix the memory leak issue

### PR DESCRIPTION
**Pull request by apepkuss**
This PR is to fix the memory leak issue reported in #2370.

### CODE REVIEW

**Comment by apepkuss** [comment]
@hydai Could you please help review this PR? Thanks!

File: bindings/rust/wasmedge-sys/examples/store_get_named_module.rs [hunk 1]
```diff
@@ -62,10 +62,12 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     // register a wasm module from a in-memory wasm bytes.
     let extern_module = Loader::create(Some(&config))?.from_bytes(wasm_bytes)?;
     Validator::create(Some(&config))?.validate(&extern_module)?;
-    executor.register_named_module(&mut store, &extern_module, "extern")?;
+    let instance = executor.register_named_module(&mut store, &extern_module, "extern")?;
 
     // get the module named "extern"
-    let instance = store.module("extern")?;
+    // let instance = store.module("extern")?;
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/examples/store_get_named_module.rs:68]: Not sure why we need to keep this comment?

**Review by hydai — APPROVED** [review]

File: bindings/rust/wasmedge-sys/examples/store_get_named_module.rs (hunk 1 above)
Comment (hydai) [inline bindings/rust/wasmedge-sys/examples/store_get_named_module.rs:68]: Not sure why we need to keep this comment?

**Review by hydai — CHANGES_REQUESTED** [review]

File: bindings/rust/wasmedge-sys/examples/store_get_named_module.rs (hunk 1 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/examples/store_get_named_module.rs:68]: Accept. I'll refactor this part. Thanks!

**Review by hydai — APPROVED** [review]

**Comment by apepkuss** [comment]
@hydai Thanks for the review!

**Pull request merged**